src/theory/term_registration_visitor.h

```diff
@@ -98,11 +98,15 @@ class PreRegisterVisitor {
    * @param visitedTheories The theories that have already preregistered current
    * @param current The term to preregister
    * @param parent The parent term of current
+   * @param preregTheories The theories that have already preregistered current.
+   * If there is no theory sharing, this coincides with visitedTheories.
+   * Otherwise, visitedTheories may be a subset of preregTheories.
```

Comment:
How exactly is `visitedTheories` different from `preregTheories`? Can't we merge them into one?

Reply:
When we in a logic with >1 theory, `visitedTheories` is the set of theories that occur in a literal, whereas preregTheories is the set of theories that have preregistered the term.

For example, if we processed `(= (f (str.len x)) 0)` and then later `(> (str.len x) 0)`, then preregistered theories for (str.len x) may be { UF, STRINGS, INT } whereas visited theories for the latter literal does not include UF.